### PR TITLE
[IMP] models: remove support for _constraints

### PR DESCRIPTION
**Pull request by mart-e**
Simply leave a warning in case a model uses a non-empty attribute
`_constraints`.

### CODE REVIEW

File: odoo/tools/translate.py [hunk 1]
```diff
@@ -935,28 +922,12 @@ def push_translation(module, type, name, id, source, comments=None):
...
-            constraints = getattr(cls, '_local_constraints', [])
-            for constraint in constraints:
-                push_constraint_msg(module, 'constraint', model._name, constraint[1])
-
     cr.execute(query_models, query_param)
 
     for (_, model, module) in cr.fetchall():
         if model not in env:
             _logger.error("Unable to find object %r", model)
             continue
```
Comment (Elkasitu) [inline odoo/tools/translate.py]: maybe this could be deleted as well? saves 1 query

**Comment by mart-e** [comment]
@robodoo r+

File: odoo/models.py [hunk 2]
```diff
@@ -1097,37 +1093,8 @@ def _log(base, record, field, exception):
...
-                if extra_error:
-                    res_msg += "\n\n%s\n%s" % (_('Error details:'), extra_error)
-                errors.append(res_msg)
-        if errors:
-            raise ValidationError('\n'.join(errors))
-
-        # new-style constraint methods
         for check in self._constraint_methods:
-            if set(check._constrains) & field_names:
+            if not field_names.isdisjoint(check._constrains):
```
Comment (mreficent) [inline odoo/models.py]: shouldn't be `if not field_names.isdisjoint(set(check._constrains)):`? 🤔

Comment (Elkasitu) [reply odoo/models.py]: [no](https://docs.python.org/3/library/stdtypes.html?highlight=isdisjoint#frozenset.isdisjoint)

Comment (mreficent) [reply odoo/models.py]: But I mean, we should make sure `check._constrains` is a `set` before that `if`, right?

Comment (Elkasitu) [reply odoo/models.py]: Why? `isdisjoint` [accepts any kind of iterable as argument](https://github.com/python/cpython/blob/3e54b575313c64f541e98216ed079fafed01ff5d/Objects/setobject.c#L1393)

Comment (mreficent) [reply odoo/models.py]: ah, ok! 😅 thanks

**Pull request closed without merging**